oaknational/Oak-Web-Application
# fix: handle nested non-primitives

### Job Story

Lessons being excluded in unitlisting are not being excluded in canonical

Eg. 

https://www.thenational.academy/pupils/lessons/chapter-1-old-majors-speech/overview

### Implementation details

changed object comparison to string comparison with json stringify src/utils/getIntersection.ts
Additional unit tests added for coverage

## How to test

There currently isn't a great way to test this due to lack of production data . However , you can check that current canonical lessons are still rendered with the link below :

https://deploy-preview-2963--oak-web-application.netlify.thenational.academy/pupils/lessons/applying-trigonometry-6mup2c/overview

### Acceptance criteria

- [x]  canonical exclusions now applied
- [x]  intersections with non-primitives handled
